sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py

```diff
@@ -79,10 +80,32 @@ def get_token(self, *scopes):
 
         # First try to get a cached access token or if a refresh token is cached, redeem it for an access token.
         # Failing that, acquire a new token.
-        app = self._app  # type: msal.ConfidentialClientApplication
+        app = self._get_app()
         result = app.acquire_token_silent(scopes, account=None) or app.acquire_token_for_client(scopes)
 
         if "access_token" not in result:
             raise ClientAuthenticationError(message="authentication failed: {}".format(result.get("error_description")))
 
         return AccessToken(result["access_token"], now + int(result["expires_in"]))
+
+    def _get_app(self):
+        # type: () -> msal.ConfidentialClientApplication
+        if not self._msal_app:
+            self._msal_app = self._create_app(msal.ConfidentialClientApplication)
+        return self._msal_app
+
+
+class PublicClientCredential(MsalCredential):
+    """Wraps an MSAL PublicClientApplication with the TokenCredential API"""
+
+    def __init__(self, **kwargs):
+        # type: (Any) -> None
+        super(PublicClientCredential, self).__init__(
+            authority="https://login.microsoftonline.com/" + kwargs.pop("tenant", "organizations"), **kwargs
+        )
+
+    def _get_app(self):
+        # type: () -> msal.PublicClientApplication
+        if not self._msal_app:
+            self._msal_app = self._create_app(msal.PublicClientApplication)
+        return self._msal_app
```

Comment:
This is only intended to ever be used as base class (since it's lacking the `get_token` method), correct? Maybe `PublicClientCredentialBase`?

Reply:
Correct, that's a good point. I added an abstract `get_token`.